esi_leap/tests/api/controllers/test_types.py

```diff
@@ -0,0 +1,22 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+# Borrowed from Ironic
+
+
+class test_collection(unittest.TestCase):
+
+    self.collection={'t1', 't2', 't3'}

```

Comment:
There's no need to set self.collection unless you plan on using it in multiple places. I would use a setUp method - something like:

   def setUp():
      self.test_collection = types.Collection()
      self.test_collection._type = 'stuff'
      self,test_collection.stuff = [1, 2, 3, 4]

(This isn't the exact right syntax for setUp; look it up in ironic tests to see how it should be coded.)

Afterwards, test_has_next can use has_next(self.test_collection, 3)